### PR TITLE
[JENKINS-61806] Form validation for TimeZoneProperty.timeZoneName

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/jenkins/pull/4748#issuecomment-654367061 seemed to have a positive reception.

tz

### Proposed changelog entries

* When configuring your **User Defined Time Zone**, you will now see hints about the current time on the server’s zone and, if not selecting **Default**, in the selected zone.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/hudson/model/TimeZoneProperty/config.properties
Comment (jglick) [inline core/src/main/resources/hudson/model/TimeZoneProperty/config.properties:21]: Was not displayed anywhere that I could see, so moving to inline help.

File: core/src/main/resources/hudson/model/Messages.properties
Comment (timja) [inline core/src/main/resources/hudson/model/Messages.properties:407]: what's di referring to here? 'proposed_di'

**Review by timja — APPROVED** [review]
checked via core pr tester

File: core/src/main/resources/hudson/model/Messages.properties
Comment (res0nance) [reply core/src/main/resources/hudson/model/Messages.properties:407]: It says `in` here twice, as well as in the code

**Review by res0nance — APPROVED** [review]

File: core/src/main/resources/hudson/model/Messages.properties
Comment (jglick) [reply core/src/main/resources/hudson/model/Messages.properties:407]: Just abbreviations of the English text, generated with an IDE plugin.

Comment (oleg-nenashev) [inline core/src/main/resources/hudson/model/Messages.properties:407]:
```suggestion
TimeZoneProperty.current_time_on_server_in_proposed_di=Current time on server in {0}: {1}; in proposed display zone: {2}
```

File: core/src/main/java/hudson/model/TimeZoneProperty.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/TimeZoneProperty.java:101]:
```suggestion
                return FormValidation.ok(Messages.TimeZoneProperty_current_time_on_server_in_proposed_di(TimeZone.getDefault().getDisplayName(), DateFormat.getDateTimeInstance().format(now), localTime.format(now)));
```

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me, but would be great to fix typo noticed by @res0nance before merging

**Comment by oleg-nenashev** [comment]
I will start the merge timeout, because the suggested change does not change the code in principle.

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

File: core/src/main/resources/hudson/model/Messages.properties
Comment (jglick) [reply core/src/main/resources/hudson/model/Messages.properties:407]: (not a typo, as noted above)

**Pull request merged**
